File: solutions/csharp/line-up/1/LineUp.cs
using System.Diagnostics.CodeAnalysis;

[SuppressMessage("Design", "CA1050:Declare types in namespaces", Justification = "<Pending>")]
public static class LineUp
{
    public static string Format(string name, int number)
    {
        var suffix = GetOrdinalSuffix(number);
        return $"{name}, you are the {number}{suffix} customer we serve today. Thank you!";
    }

    private static string GetOrdinalSuffix(int number) => (number % 10) switch
    {
        1 when number % 100 != 11 => "st",
        2 when number % 100 != 12 => "nd",
        3 when number % 100 != 13 => "rd",
        _ => "th"
    };
}

File: solutions/python/line-up/1/line_up.py
ORDINAL_SUFFIXES = {
    1: "st",
    2: "nd",
    3: "rd",
}

def get_ordinal_suffix(number):
    last_two_digits = number % 100
    if 11 <= last_two_digits <= 13:
        return "th"

    last_digit = number % 10
    return ORDINAL_SUFFIXES.get(last_digit, "th")

def line_up(name, number):
    ordinal_suffix = get_ordinal_suffix(number)
    return (
        f"{name}, you are the {number}{ordinal_suffix} "
        "customer we serve today. Thank you!"
    )